src/runtime/internal/dom.ts

```diff
@@ -630,6 +630,25 @@ export function query_selector_all(selector: string, parent: HTMLElement = docum
 	return Array.from(parent.querySelectorAll(selector)) as ChildNodeArray;
 }
 
+export function head_selector(nodeId: string, head: HTMLElement) {
+	const result = [];
+	let started = 0;
+	for (const node of head.childNodes) {
+		if (node.nodeType === 8 /* comment node */ && node.textContent.trim() === 'HTML_TAG_END') {
+			started -= 1;
+			result.push(node);
+		} else if (node.nodeType === 8 /* comment node */ && node.textContent.trim() === 'HTML_TAG_START') {
+			started += 1;
+			result.push(node);
+		} else if (started > 0) {
+			result.push(node);
+		} else if (node.nodeType === 1 /* element node */ && (<Element>node).getAttribute('data-svelte') === nodeId) {
+			result.push(node);
+		}
+	}
+	return result;
+}
+
```

Comment:
If you have multiple `<svelte:head>` elements, wouldn't this be selecting `{@html}` elements from another `<svelte:head>` element?

Reply:
I imagine it would, yeah. This would also be addressed if the `HtmlTag` class (or, I guess, the `HtmlTagHydration` class) were responsible for inserting this attribute on the top level elements - since each component would only be removing elements matching its unique scoping class name.